lib/uploadcare/resources/resource_list.rb

```diff
@@ -0,0 +1,71 @@
+require 'ostruct'
+
+module Uploadcare
+  class Api
+    class ResourceList
+      include Enumerable
+
+      extend Forwardable
+      def_delegator :@data, :meta
+      def_delegator :@data, :objects
+      def_delegator :objects, :[]
+
+      attr_reader :options
+
+      def initialize(api, data, options)
+        @api = api
+        @data = OpenStruct.new(
+          meta: data.reject{|k, _| k == 'results'}.freeze,
+          objects: data['results'].map{|object| to_resource(api, object)}
+        )
+        @options = options.dup.freeze
+      end
+
+      def each
+        return enum_for(:each) unless block_given?
+
+        objects.each{|object| yield object}
+
+        unless last?
+          next_page = get_next_page
+          next_page.each(&Proc.new)
+        end
+
+        self
+      ensure
+        @data = OpenStruct.new(
+          meta: next_page.meta,
+          objects: objects + next_page.objects
+        ) if next_page
+      end
+
+      # TODO: #delete and #store methods
+
+      def total
+        meta['total']
+      end
+
+      def loaded
+        objects.size
+      end
+
+      private
+      attr_reader :api
+
+      def get_next_page
```

Comment:
perhaps this should be renamed to `get_rest_of_list` or `get_list_tail` because it returns not just a page, but a representation of the rest of the list